components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt

```diff
@@ -464,13 +465,23 @@ class TabsUseCases(
     class DuplicateTabUseCase(
         private val store: BrowserStore
     ) {
+        /**
+         * Creates a duplicate of the currently selected tab (including history) and
+         * selects it if [selectNewTab] is true.
+         */
+        fun invoke(selectNewTab: Boolean): String? {
+            return store.state.selectedTab?.let {
+                this.invoke(it, selectNewTab)
+            }
+        }
+
         /**
          * Creates a duplicate of [tab] (including history) and selects it if [selectNewTab] is true.
          */
         operator fun invoke(
```

Comment:
We could also use the existing operator overload here to do the same thing. What do you think?

```kotlin
operator fun invoke(
    tab: TabSessionState? = null,
    selectNewTab: Boolean = true
) {
    val tabToDuplicate = tab ?: store.state.selectedTab ?: return // return or throw?
    val duplicate = createTab(
        url = tabToDuplicate.content.url,
        private = tabToDuplicate.content.private,
        contextId = tabToDuplicate.contextId,
        parent = tabToDuplicate,
        engineSessionState = tabToDuplicate.engineState.engineSessionState
    )
```

Reply:
I considered that, but 2 things pushed me to use the current method:

1) The 2-argument invoke is guaranteed to return `String`, but since `selectedTab` is nullable the return of the 1-argument invoke has to be `String?`. Unless it's reasonable to assume that there's always a selected tab and to just `!!` it?

2) `AddNewTabUseCase` is a precedent here as it also has 2 invoke methods with different return types.

Reply:
Yeah, those are fair points. Thanks for detailed response. :)